evm/rlp/transactions.py

```diff
@@ -160,3 +160,32 @@ class BaseShardingTransaction(rlp.Serializable):
     @property
     def hash(self):
         return keccak(rlp.encode(self))
+
+    #
+    # Validation
+    #
+    def validate(self):
+        """
+        Hook called during instantiation to ensure that all transaction
+        parameters pass validation rules.
+        """
+        if self.intrensic_gas > self.gas:
```

Comment:
Correct spelling is "intrinsic".

Reply:
@pipermerriam I will update the naming in sharding branch along with this PR and update the one in master branch with a new PR.